doc/manuals/admin/build_source.md

```diff
@@ -51,6 +51,16 @@ commands that require root privilege):
         sudo make install  # installation puts .h files in /usr/local/include and library in /usr/local/lib
         sudo ldconfig      # just in case... it doesn't hurt :)
 
+* Install mosquitto from sources (config.mk file under mosquitto-1.5/ can be modified to fine tune the build)
+
+        wget -4 http://mosquitto.org/files/source/mosquitto-1.5.tar.gz
+        tar xvf mosquitto-1.5.tar.gz
+        cd mosquitto-1.5
+        make
+        sudo make install  # installation puts .h files in /usr/local/include and library in /usr/local/lib
+        echo "/usr/local/lib" | sudo tee /etc/ld.so.conf.d/99local.conf # tell runtime loader to look into /usr/local/lib
```

Comment:
Is this really needed? The other dependency libraries (e.g. libmicrohttpd above) also install in /usr/local/lib and no other needs this...

Maybe is something specific of your local system?

Reply:
> The other dependency libraries (e.g. libmicrohttpd above) also install in /usr/local/lib and no other needs this...

Exactly, I did not need to do this for the other libraries, so either my system or the mosquitto build has an issue. 

> Maybe is something specific of your local system?

To be sure, I double checked with a fresh CentOS 7 installation and the result was the same. 

Then I checked the makefiles of the dependencies and realized that, `mongo-cxx-driver`, `libmicrohttpd` and `gmock` are using [libtool](https://www.gnu.org/software/libtool/manual/libtool.html) but mosquitto is not. Libtool creates `.la` files alongside the shared objects which are used by libtool manage the library linking.

> Is this really needed?

I think yes, in order to get mosquitto working, the system should have a `.conf` file under `/etc/ld.so.conf.d`. There are other ways to tell the runtime loader to look in a specific path for libraries but the one we have right now is considered to be best practice (as far as I know).

Changing the /usr/local prefix in `config.mk` file of mosquitto to another location before running make would also be a solution but I prefer the one above.

Please let me know if I am missing something.